### PR TITLE
XD-3456  Timestamp Task Module

### PR DESCRIPTION
**Pull request by cppwfs**
This resolves spring-cloud/spring-cloud-task-modules#2

### CODE REVIEW

File: timestamp-task/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Timestamp Job
+============================
```
Comment (ericbottard) [inline timestamp-task/README.md]: Can we use asciidoctor for this please?
Also, please do not use `=======` underlying for titles, but rather prefixes. They're easier to change via `sed` or similar if needed

File: timestamp-task/pom.xml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>org.springframework.cloud.task.module</groupId>
+	<artifactId>timestamp-task</artifactId>
+	<packaging>jar</packaging>
+	<name>timestamp-task</name>
+	<description>Spring Cloud Stream timestamp task</description>
+
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (ericbottard) [inline timestamp-task/pom.xml]: should be inherited

File: timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskApplication.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * Executes a batch job that logs a timestamp
+ *
+ * @author Glenn Renfro
+ */
+@SpringBootApplication
+@EnableBatchProcessing
```
Comment (ericbottard) [inline timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskApplication.java]: Should this annotation move to the configuration class?

File: timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskProperties.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	 * The timestamp format, "yyyy-MM-dd HH:mm:ss.SSS" by default.
+	 */
+	private String format  = "yyyy-MM-dd HH:mm:ss.SSS";
+
+	public String getFormat() {
+		return format;
+	}
+
+	public void setFormat(String format) {
+		Assert.hasText(format, "format must not be empty nor null");
```
Comment (ericbottard) [inline timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskProperties.java]: Should use jsr303 validation for that, not Assert

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	<url>http://projects.spring.io/spring-xd/</url>
+	<organization>
+		<name>Pivotal Software, Inc.</name>
+		<url>http://www.spring.io</url>
+	</organization>
+	<parent>
+		<groupId>org.springframework.cloud</groupId>
+		<artifactId>spring-cloud-stream-parent</artifactId>
+		<version>1.0.0.BUILD-SNAPSHOT</version>
+	</parent>
```
Comment (mminella) [inline pom.xml:19]: What are we getting from this?  I'm not saying we won't need it in the future, but I want to be cautious about how we tie these two projects together...

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+		<!-- tasks in alphabetical order -->
+		<module>timestamp-task</module>
+	</modules>
+	<build>
+		<plugins>
+			<plugin>
+				<!--skip deploy (this is just a test module) -->
+				<artifactId>maven-deploy-plugin</artifactId>
+				<configuration>
+					<!-- <skip>true</skip> -->
```
Comment (mminella) [inline pom.xml]: Remove commented out code

File: timestamp-task/README.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
+= Timestamp Job
+
+This module executes a Spring Batch Job Tasklet that logs a timestamp.
+
+== Requirements:
+
+* Java 7 or Above
+* Redis running on localhost
```
Comment (mminella) [inline timestamp-task/README.adoc]: What does this need Redis for?

File: timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTask.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	private TimestampTaskProperties config;
+
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	@Bean
+	protected Tasklet tasklet() {
+		return new Tasklet() {
+			@Override
+			public RepeatStatus execute(StepContribution contribution,
+										ChunkContext context) {
```
Comment (mminella) [inline timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTask.java]: Odd indent.

File: timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskApplication.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Executes a batch job that logs a timestamp
+ *
+ * @author Glenn Renfro
+ */
+@SpringBootApplication
+public class TimestampTaskApplication {
+	public static void main(String[] args) throws Exception {
+		SpringApplication.exit(SpringApplication.run(TimestampTaskApplication.class, args));
```
Comment (mminella) [inline timestamp-task/src/main/java/org/springframework/cloud/task/module/timestamp/TimestampTaskApplication.java]: Do we need the `SpringApplication.exit`?

File: timestamp-task/pom.xml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-starter-config</artifactId>
+			<version>${cloud-config.version}</version>
+		</dependency>
```
Comment (mminella) [inline timestamp-task/pom.xml]: Does it make sense to move this dependency up a level?  What modules would _not_ need it?

**Comment by cppwfs** [comment]
https://github.com/spring-cloud/spring-cloud-task-modules/pull/13 supersedes this PR.

**Pull request closed without merging**